components/db/testimony/usePublishedTestimonyListing.ts

```diff
@@ -91,16 +104,66 @@ async function listTestimony(
   limitCount: number,
   startAfterKey: unknown | null
 ): Promise<Testimony[]> {
-  const testimonyRef = collectionGroup(firestore, "publishedTestimony")
-  const result = await getDocs(
-    nullableQuery(
-      testimonyRef,
-      ...getWhere(refinement),
-      where("court", "==", currentGeneralCourt),
-      orderBy("publishedAt", "desc"),
-      limit(limitCount),
-      startAfterKey !== null && startAfter(startAfterKey)
-    )
-  )
-  return result.docs.map(d => d.data() as Testimony)
+  const client = createClient()
+  const publishedBool = refinement.published === "true"
+
+  let query = { }
+
+  if (!publishedBool && refinement.billId && refinement.uid) {
+      const result = await getBillTestimony(refinement.uid, refinement.billId)
+
+      return [result.draft] as Testimony[]
+  }
+
+  if (refinement.billId && refinement.uid) {
+    query = {
+      q: `${refinement.uid}, ${refinement.billId}`,
+      query_by: "authorUid, billId",
+    }
+    console.log(1, query)
+  } else if (refinement.billId) {
+    console.log(2)
+    query = {
+      q: refinement.billId,
+      query_by: "billId"
+    }
+  } else if (refinement.uid) {
+    console.log(3)
+    query = {
+      q: refinement.uid,
+      query_by: "authorUid"
+    }
+  } else {
+    query = {
+      q: "*",
+      query_by: "billId"
+    }
+  }
+
+
+  const data = await client
+    .collections("publishedTestimony")
+    .documents()
+    .search(query)
+
+  console.log('Data', data)
+  return data.hits
+    ? data.hits.map(({ document }) => ({
```

Comment:
I think we can cast these docs directly to TestimonySearchRecord, since they come straight from typesense